conformance/data_model/float.ion

```diff
@@ -0,0 +1,288 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+TESTING NOTES:
+
+It is assumed that floats are largely handled by a standard library with reliable test covereage.
+Do not use this test suite as the only standard for any home-spun floating point number implementation.
+
+In IEEE 754 interchange formats, binary format NaNs are represented with the exponential field filled with ones (like
+infinity values), and ANY non-zero number in the significand field. These test cases use the quiet and signalling NaN
+encodings as specified in IEEE-754-2019, although there areother legal representations of NaN.
+ */
+
+(ion_1_0 "Ion 1.0 binary"
+         (then "zero encoded as"
+               (each "f0"  (binary "40")
+                     "f32" (binary "44 00 00 00 00")
+                     "f64" (binary "48 00 00 00 00 00 00 00 00")
+                     (denotes (Float "0.0"))))
+         (then "negative zero encoded as"
+               (each "f32" (binary "44 80 00 00 00")
+                     "f64" (binary "48 80 00 00 00 00 00 00 00")
+                     (denotes (Float "-0.0"))))
+         (then "1.0 encoded as"
+               (each "f32" (binary "44 3F 80 00 00")
+                     "f64" (binary "48 3F F0 00 00 00 00 00 00")
+                     (denotes (Float "1.0"))))
+         (then "-1.0 encoded as"
+               (each "f32" (binary "44 BF 80 00 00")
+                     "f64" (binary "48 BF F0 00 00 00 00 00 00")
+                     (denotes (Float "-1.0"))))
+         (then "an ordinary non-integral number encoded as"
+               (each "f32" (binary "44 40 C4 00 00")
+                     "f64" (binary "48 40 18 80 00 00 00 00 00")
+                     (denotes (Float "6.125"))))
+         (then "a subnormal"
+               (then "f32 value"
+                     (binary "44 00 00 00 01")
+                     (denotes (Float "1.401298464324817E-45")))
+               (then "f64 value"
+                     (binary "48 00 00 00 00 00 00 00 01")
+                     (denotes (Float "4.9406564584124654e-324"))))
```

Comment:
```suggestion
                     (denotes (Float "5e-324"))))
```

Reply:
This one, and the others you mentioned, are subnormal numbers, which have this much precision.

Reply:
Oh, shoot. I was the one who missed something. I was thrown off by something else that I ran into.